store/tikv/txn.go

```diff
@@ -185,20 +184,41 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
 		connID = val.(uint64)
 	}
 	committer, err := newTwoPhaseCommitter(txn, connID)
-	if err != nil {
+	if err != nil || committer == nil {
 		return errors.Trace(err)
 	}
-	if committer == nil {
-		return nil
+
+	defer func() {
+		if err == nil {
+			txn.commitTS = committer.commitTS
+		}
+	}()
+	// latches disabled
+	if txn.store.txnLatches == nil {
+		err = committer.executeAndWriteFinishBinlog(ctx)
+		return errors.Trace(err)
 	}
-	err = committer.execute(ctx)
-	if err != nil {
-		committer.writeFinishBinlog(binlog.BinlogType_Rollback, 0)
+
+	// latches enabled
+	// for transactions not retryable, commit directly.
+	if !sessionctx.GetRetryable(ctx) {
+		err = committer.executeAndWriteFinishBinlog(ctx)
+		txn.store.txnLatches.RefreshCommitTS(committer.keys, committer.startTS)
```

Comment:
Why use the `startTS` to refresh? Do we need to refresh if an error occurred when commit?